product_configurator/models/product_config.py

```diff
@@ -1224,13 +1224,25 @@ def values_available(
             config_lines = product_tmpl.config_line_ids.filtered(
                 lambda l: attr_val_id in l.value_ids.ids
             )
-            domains = config_lines.mapped("domain_id").compute_domain()
-            avail = self.validate_domains_against_sels(domains, value_ids, custom_vals)
-            if avail:
-                avail_val_ids.append(attr_val_id)
-            elif attr_val_id in value_ids:
-                value_ids.remove(attr_val_id)
-
+            if len(config_lines) > 1:
+                for config in config_lines:
+                    domain = config.domain_id.compute_domain()
+                    avalable = self.validate_domains_against_sels(
+                        domain, value_ids, custom_vals
+                    )
+                    if avalable:
+                        avail_val_ids.append(attr_val_id)
+                    elif attr_val_id in value_ids:
+                        value_ids.remove(attr_val_id)
+            else:
```

Comment:
I'm not sure I understand the logic here. If there is only one config_line then we do not need mapped and the code above would work and we could just keep the loop for both one or multiple lines. Second thing is compute_domain takes care of all domains, be it called individually for each one or bundled together. Lastly there is an implicit logic of chaining all the conditions in the domain together with an OR operator which should be in theory the same as calling them separately. I have to look more into it to be sure but I have a feeling the problem is here https://github.com/OCA/product-configurator/blob/14.0/product_configurator/models/product_config.py#L1178 instead of the domain computation.

Reply:
@PCatinean Thank you for your input and I understand what you are saying and I'm open for your suggestions if there is a better approach. 

Here were my original thoughts and understanding of the code (which could be wrong, it's a complicated code set for me :)). I'll spell it out here for my own sanity and you can tell me if I'm wrong or looking at this differently:

- The 'compute_domain' method seems to puts together the domains from the line items on the restrictions (aka config lines) to create the domain for that particular config line. 
- The 'vaildate_domains_against_sels' checks if the provided domain(s) based on the values given and returns true/false.
- The 'values_available' pieces together the available values that the user can select, based on the config lines and their domains. Before my changes, it originally would look at all the domains together and if one failed then remove the value even if it exists in the another config line.

What my additions do is basically make it so if there are more than one config line for the same attribute, it will cycle through evaluating each one individually and then changing the available values accordingly. Otherwise it just does what the original code did.
I tried to make it a single for loop to simplify the function but it doesn't want to work. Theoretically it should work but I'm not exactly sure why. It would tell me that I needed to select at least one attribute value when trying to start configuring a product. I'm sure I had an error in my code or something that caused this, I just don't know at this time.

When first looking at the code I did see where you reference to the "AND" vs "OR" in the other methods and that might be a better solution but I'm not sure how it would do that since the compute domain is basically looking at the config line domains and not looking at the config lines themselves so I'm open for more discussion/suggestions on this.